docs/src/main/asciidoc/platform.adoc

```diff
@@ -0,0 +1,149 @@
+
+[source,xml]
+----
+<plugin>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-platform-descriptor-json-plugin</artifactId>
+  <version>${quarkus.version}</version> <1>
+  <executions>
+    <execution>
+      <phase>process-resources</phase>
+      <goals>
+        <goal>generate-extensions-json</goal> <2>
+      </goals>
+    </execution>
+  </executions>
+  <configuration>
+    <bomGroupId>${quarkus.platform.group-id}</bomGroupId> <3>
+    <bomArtifactId>${quarkus.platform.artifact-id}</bomArtifactId> <4>
+    <bomVersion>${quarkus.platform.version}</bomVersion> <5>
+    <overridesFile>${overridesfile}</overridesFile> <6>
+    <resolveDependencyManagement>true</resolveDependencyManagement> <7>
+  </configuration>
+</plugin>
+----
+
+<1> the version of the `quarkus-platform-descriptor-json-plugin`
+<2> `generate-extensions-json` is the goal generating the platform descriptor
+<3> the `groupId` of the platform BOM
+<4> the `artifactId` of the platform BOM
+<5> the `version` of the platform BOM
+<6> this parameter is optional, it allows to override some metadata from the Quarkus extension descriptors found in every runtime extension artifact from which the platform descriptor is generated
+<7> this parameter is also optional and defaults to false. It has to be set to true in case the platform BOM *is not generated* and *is not flattened*. Which for example is the case for `io.quarkus:quarkus-bom`.
+
+[[platform-properties]]
+=== Quarkus Platform Properties
+
+A Quarkus platform may provide its own default values for some of the configuration options.
+
+Quarkus is using https://github.com/eclipse/microprofile-config[Eclipse MicroProfile Config^] for wiring application configuration. A Quarkus platform may be used as another source of configuration in the hierarchy of the configuration sources dominated by the application's `application.properties`.
+
+To provide platform-specific defaults, the platform needs to include a dependency version constraint in its BOM for a properties artifact whose coordinates follow the following naming convention:
+
+* the `groupId` of the properties artifact should match the `groupId` of the corresponding Quarkus Platform BOM;
+* the `artifactId` of the properties artifact should be the `artifactId` of the corresponding Quarkus Platform BOM with the `-quarkus-platform-properties` suffix;
+* the `classifier` of the descriptor artifact should be left empty/null;
+* the `type` of the descriptor artifact should be `properties`;
+* the `version` of the descriptor artifact should match the `version` of the  corresponding Quarkus Platform BOM.
+
+The properties artifact itself is expected to be a traditional `properties` file that will be loaded into an instance of `java.util.Properties` class.
+
+IMPORTANT: At this point, platform properties are only allowed to provide the default values to a restricted set of properties. Property names in the platform properties file must be prefixed with the `platform.` suffix.
```

Comment:
```suggestion
IMPORTANT: At this point, platform properties are only allowed to provide the default values of a restricted set of properties. Property names in the platform properties file must be prefixed with the `platform.` suffix.
```
Reads better IMHO.

Reply:
Perhaps "for" will be even better?

Reply:
I think "provide ... for" sounds better. And probably it should be "a restricted set of configuration options".

Reply:
I re-wrote it as
-IMPORTANT: At this point, platform properties are only allowed to provide the default values to a restricted set of properties. Property names in the platform properties file must be prefixed with the `platform.` suffix.
+IMPORTANT: At this point, platform properties are only allowed to provide the default values for a restricted set of configuration options. The property names in the platform properties file must be prefixed with the `platform.` suffix.

Thanks for the review!